docs/reference/snapshot-restore/apis/create-snapshot-api.asciidoc

```diff
@@ -0,0 +1,193 @@
+
+IMPORTANT: While a snapshot of a particular shard is being created, this shard cannot be moved to another node. Relocating a shard during the snapshot process can interfere with rebalancing and allocation filtering. {es} can move a shard to another node (according to the current allocation filtering settings and rebalancing algorithm) only after the snapshot process completes.
+
+Besides creating a copy of each data stream and index, the snapshot process can also store global cluster metadata, including persistent cluster settings and templates. The transient settings and registered snapshot repositories are not stored as part of the snapshot.
+
+[[create-snapshot-api-path-params]]
+==== {api-path-parms-title}
+
+`<repository>`::
+(Required, string)
+Name of the repository to create a snapshot in.
+
+`<snapshot>`::
+(Required, string)
+Name of the snapshot to create. This name must be unique in the snapshot repository.
+
+[role="child_attributes"]
+[[create-snapshot-api-request-body]]
+==== {api-request-body-title}
+
+`ignore_unavailable`::
+(Optional, boolean)
+If `false`, the request returns an error for any data stream or index that is missing or closed. Defaults to `false`.
++
+If `true`, the request ignores data streams and indices in `indices` that are missing or closed.
+
+`indices`::
+(Optional, string)
+A comma-separated list of data streams and indices to include in the snapshot.
+<<multi-index,Multi-index syntax>> is supported.
++
+By default, a snapshot includes all data streams and indices in the cluster. If this
+argument is provided, the snapshot only includes the specified data streams and clusters.
+
+`include_global_state`::
++
+--
+(Optional, boolean)
+If `true`, the current cluster state is included in the snapshot.
+Defaults to `true`.
+
+The cluster state includes:
+
+* Persistent cluster settings
+* Index templates
+* Legacy index templates
+* Ingest pipelines
+* {ilm-init} lifecycle policies
+--
++
+IMPORTANT: By default, the entire snapshot will fail if one or more indices included in the snapshot do not have all primary shards available. You can change this behavior by setting <<create-snapshot-api-partial,`partial`>> to `true`.
+
+include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=master-timeout]
+
+`metadata`::
+(Optional, string)
+Attaches arbitrary metadata to the snapshot, such as a record of who took the snapshot, why it was taken, or any other useful data. Metadata must be less than 1024 bytes.
+
+[[create-snapshot-api-partial]]
+`partial`::
+(Optional, boolean)
+If `false`, the entire snapshot will fail if one or more indices included in the snapshot do not have all primary shards available. Defaults to `false`.
++
+If `true`, allows taking a partial snapshot of indices with unavailable shards.
+
+`settings`::
+(Optional, object)
+Sets snapshot settings for the repository in JSON or YAML format. See <<put-snapshot-repo-api-settings-param,`settings`>> in the Put snapshot repository API documentation.
```

Comment:
I tested this parameter, and it seems weird... you can throw any arbitrary object in there and it accepts it.

@original-brownbear Any thoughts? I wonder if this is just a param that never got used. If so, we should maybe look at removing it.

In any case, I don't think this definition is accurate or makes sense. You shouldn't be able to overwrite snapshot repo settings from a snapshot.

If this is just an unused param, I wonder if we should A) not document it or B) document it with a warning that it doesn't do anything. I'd hate to see a user throw values in here expecting something to happen if it does nothing.

Reply:
> If this is just an unused param, I wonder if we should A) not document it or B) document it with a warning that it doesn't do anything. I'd hate to see a user throw values in here expecting something to happen if it does nothing.

Sorry for the delay here. You're right ... this parameter doesn't do anything at the moment (I'll look into removing it). IMO, we probably should just not document it. Was it ever documented in the first place even?

Reply:
https://github.com/elastic/elasticsearch/pull/58661

Reply:
Thanks @jrodewig for testing this parameter, and @original-brownbear for confirming that we can remove it. With that parameter removed, I think that this PR is ready to merge.

Reply:
Agree. Thanks for taking this one to the finish line @lockewritesdocs.